File: fuel/app/config/js.php
<?php
$webpack = \Config::get('materia.urls.js_css');
$vendor = \Config::get('materia.urls.js_css').'vendor/';
$cdnjs = '//cdnjs.cloudflare.com/ajax/libs/';

return [
	'groups' => [
		'login'      => [$webpack.'js/login.js'],
		'profile'    => [$webpack.'js/profile.js'],
		'settings'   => [$webpack.'js/settings.js'],
		'support'    => [$webpack.'js/support.js'],
		'user_admin' => [$webpack.'js/user-admin.js'],
		'widget_admin' => [$webpack.'js/widget-admin.js'],
		'materia'    => [$webpack.'js/materia.js'],
		'homepage'   => [$webpack.'js/homepage.js'],
		'catalog'    => [$webpack.'js/catalog.js'],
		'my_widgets' => [$webpack.'js/my-widgets.js'],
		'detail'     => [$webpack.'js/detail.js'],
		'playpage'   => [$webpack.'js/player-page.js'],
		'createpage' => [$webpack.'js/creator-page.js'],
		'scores'     => [$webpack.'js/scores.js'],
		'guides'     => [$webpack.'js/guides.js'],
		'retired'    => [$webpack.'js/retired.js'],
		'no_attempts'=> [$webpack.'js/no-attempts.js'],
		'draft_not_playable' => [$webpack.'js/draft-not-playable.js'],
		'no_permission' => [$webpack.'js/no-permission.js'],
		'embedded_only' => [$webpack.'js/embedded-only.js'],
		'pre_embed'  => [$webpack.'js/pre-embed-placeholder.js'],
		'help'       => [$webpack.'js/help.js'],
		'404'        => [$webpack.'js/404.js'],
		'500'        => [$webpack.'js/500.js'],
		'media'      => [$webpack.'js/media.js'],
		'qset_history' => [$webpack.'js/qset-history.js'],
		'post_login' => [$webpack.'js/lti-post-login.js'],
		'select_item' => [$webpack.'js/lti-select-item.js'],
		'open_preview' => [$webpack.'js/lti-open-preview.js'],
		'error_general' => [$webpack.'js/lti-error.js'],
		'react'      => [
			'//unpkg.com/react@16.13.1/umd/react.development.js',
			'//unpkg.com/react-dom@16.13.1/umd/react-dom.development.js',
			$webpack.'js/include.js'
		],
		'question-importer' => [$webpack.'js/question-importer.js']
	]
];

File: app/materia/settings/js.py
import os

# JS group definition configs

JS_BASEURL = os.environ.get("JS_BASEURL", "dist/js/")

JS_GROUPS = {
    "main": [JS_BASEURL + "homepage.js"],
    "my-widgets": [JS_BASEURL + "my-widgets.js"],
    "help": [JS_BASEURL + "help.js"],
    "catalog": [JS_BASEURL + "catalog.js"],
    "community-library": [JS_BASEURL + "community-library.js"],
    "detail": [JS_BASEURL + "detail.js"],
    "player": [JS_BASEURL + "player-page.js"],
    "creator": [JS_BASEURL + "creator-page.js"],
    "login": [JS_BASEURL + "login.js"],
    "closed": [JS_BASEURL + "closed.js"],
    "draft-not-playable": [JS_BASEURL + "draft-not-playable.js"],
    "retired": [JS_BASEURL + "retired.js"],
    "profile": [JS_BASEURL + "profile.js"],
    "settings": [JS_BASEURL + "settings.js"],
    "scores": [JS_BASEURL + "scores.js"],
    "media": [JS_BASEURL + "media.js"],
    "widget_admin": [JS_BASEURL + "widget-admin.js"],
    "user_admin": [JS_BASEURL + "user-admin.js"],
    "instance_admin": [JS_BASEURL + "support.js"],
    "no-permission": [JS_BASEURL + "no-permission.js"],
    "no-attempts": [JS_BASEURL + "no-attempts.js"],
    "pre-embed": [JS_BASEURL + "pre-embed-placeholder.js"],
    "embedded-only": [JS_BASEURL + "embedded-only.js"],
    "post-login": [JS_BASEURL + "lti-post-login.js"],
    "select-item": [JS_BASEURL + "lti-select-item.js"],
    "open-preview": [JS_BASEURL + "lti-open-preview.js"],
    "lti-error": [JS_BASEURL + "lti-error.js"],
    "404": [JS_BASEURL + "404.js"],
    "500": [JS_BASEURL + "500.js"],
    "guides": [JS_BASEURL + "guides.js"],
    "qset-history": [JS_BASEURL + "qset-history.js"],
    "qset-generator": [JS_BASEURL + "qset-generator.js"],
}
